### PR TITLE
Add support for encrypting PKCS#12 stores with a password

### LINKED ISSUE #313: Allow users to override PKCS#12 password
Some applications (HDFS, grumble grumble) refuse to load certificates from keystores with no password.

### PR DESCRIPTION
**Pull request by nightkr**
# Description

Fixes #313.

We add a new volume parameter `secrets.stackable.tech/format.compatibility.tls-pkcs12.password` for controlling this. The password should not be considered private (PVC manifests are generally public!), this is purely for compatibility with *certain* broken apps.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [ ] CRD changes approved
- [x] Helm chart can be installed and deployed operator works
- [x] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

**Comment by nightkr** [comment]
I'm not sure if this is something that we want to document, since this is more of a desperate "feature of last resort".

File: rust/operator-binary/src/backend/mod.rs [hunk 1]
```diff
@@ -72,6 +72,17 @@ pub struct SecretVolumeSelector {
         deserialize_with = "SecretVolumeSelector::deserialize_str_vec"
     )]
     pub kerberos_service_names: Vec<String>,
+
+    /// A password that the TLS PKCS#12 keystore should be protected using
```
Comment (maltesander) [inline rust/operator-binary/src/backend/mod.rs:76]:
```suggestion
    /// A password to protect the generated TLS PKCS#12 keystores
```

**Review by maltesander — DISMISSED** [review]
LGTM! Lets wait what the arch meeting brings.

File: rust/operator-binary/src/backend/mod.rs (hunk 1 above)
Comment (nightkr) [reply rust/operator-binary/src/backend/mod.rs:76]: On second thought, I don't like "protect" at all in this case. We're not protecting anything, just adding yet another box ticking exercise.

Comment (maltesander) [reply rust/operator-binary/src/backend/mod.rs:76]: Yeah im fine with another wording.

Comment (siegfriedweber) [reply rust/operator-binary/src/backend/mod.rs:76]: In my opinion, this sentence is grammatically incorrect. We should fix it before merging. What about "The passphrase used to encrypt the TLS PKCS#12 keystore"?

**Review by siegfriedweber — CHANGES_REQUESTED** [review]
In general okay after the comment for `compat_tls_pkcs12_password` is fixed.

I would not emphasize it so much that the password is used for compatibility reasons. It is an option (and not a compatibility option) which is used for compatibility and not too useful on its own.

File: rust/operator-binary/src/backend/mod.rs (hunk 1 above)
Comment (nightkr) [reply rust/operator-binary/src/backend/mod.rs]: be3ab3ce2675b137512394d8da9d8217c9316f33

I stuck to password for consistency.

**Comment by nightkr** [comment]
It *only* exists for compatibility, and provides no useful security improvement on its own.

I absolutely want to avoid misleading people around this...

**Review by siegfriedweber — APPROVED** [review]
LGTM

**Review by sbernauer — APPROVED** [review]

**Pull request merged**